gobblin-kafka/src/main/java/gobblin/kafka/schemareg/LiKafkaSchemaRegistry.java

```diff
@@ -0,0 +1,247 @@
+
+import gobblin.configuration.ConfigurationKeys;
+import gobblin.kafka.serialize.MD5Digest;
+import gobblin.util.AvroUtils;
+
+
+/**
+ * Integration with LinkedIn's implementation of a schema registry that uses md5-hash for schema ids.
+ */
+public class LiKafkaSchemaRegistry implements KafkaSchemaRegistry<MD5Digest, Schema> {
+
+
+  private static final Logger LOG = LoggerFactory.getLogger(LiKafkaSchemaRegistry.class);
+  private static final String GET_RESOURCE_BY_ID = "/id=";
+  private static final String GET_RESOURCE_BY_TYPE = "/latest_with_type=";
+  private static final String SCHEMA_ID_HEADER_NAME = "Location";
+  private static final String SCHEMA_ID_HEADER_PREFIX = "/id=";
+
+  private final GenericObjectPool<HttpClient> httpClientPool;
+  private final String url;
+
+  /**
+   * @param props properties should contain property "kafka.schema.registry.url", and optionally
+   * "kafka.schema.registry.max.cache.size" (default = 1000) and
+   * "kafka.schema.registry.cache.expire.after.write.min" (default = 10).
+   */
+  public LiKafkaSchemaRegistry(Properties props) {
+    //super(props);
+    Preconditions.checkArgument(props.containsKey(KafkaSchemaRegistryConfigurationKeys.KAFKA_SCHEMA_REGISTRY_URL),
+        String.format("Property %s not provided.", KafkaSchemaRegistryConfigurationKeys.KAFKA_SCHEMA_REGISTRY_URL));
+
+    this.url = props.getProperty(KafkaSchemaRegistryConfigurationKeys.KAFKA_SCHEMA_REGISTRY_URL);
+
+    int objPoolSize =
+        Integer.parseInt(props.getProperty(ConfigurationKeys.KAFKA_SOURCE_WORK_UNITS_CREATION_THREADS,
+            "" + ConfigurationKeys.KAFKA_SOURCE_WORK_UNITS_CREATION_DEFAULT_THREAD_COUNT));
+    LOG.info("Create HttpClient pool with size " + objPoolSize);
+
+    GenericObjectPoolConfig config = new GenericObjectPoolConfig();
+    config.setMaxTotal(objPoolSize);
+    config.setMaxIdle(objPoolSize);
+    this.httpClientPool = new GenericObjectPool<>(new HttpClientFactory(), config);
+  }
+
+
+  @Override
+  public Schema getById(MD5Digest id)
+      throws IOException, SchemaRegistryException {
+    return fetchSchemaByKey(id);
+  }
+
+  /**
+   *
+   * @return whether this implementation of the schema registry has an internal cache
+   */
+  @Override
+  public boolean hasInternalCache() {
+    return false;
+  }
+
+  /**
+   * Get the latest schema of a topic.
+   *
+   * @param topic topic name
+   * @return the latest schema
+   * @throws SchemaRegistryException if failed to retrieve schema.
+   */
+  @Override
+  public Schema getLatestSchema(String topic) throws SchemaRegistryException {
+    String schemaUrl = this.url + GET_RESOURCE_BY_TYPE + topic;
+
+    LOG.debug("Fetching from URL : " + schemaUrl);
+
+    GetMethod get = new GetMethod(schemaUrl);
+
+    int statusCode;
+    String schemaString;
+    HttpClient httpClient = this.borrowClient();
+    try {
+      statusCode = httpClient.executeMethod(get);
+      schemaString = get.getResponseBodyAsString();
```

Comment:
If statusCode is not SC_OK, we don't need to fetch response body. So we can move this to line 128